### PR TITLE
Add RTFHandler unittest

### PR DESCRIPTION
**Pull request by javierag**
This is an adaptation of jagarcia/unittest to make it ready for mergin.

It is needed as requisite of jag/rtf-parse-fix-unicode

### CODE REVIEW

**Comment by javierag** [comment]
There will be changes on method names but they will be in ag/rtf-parse-fix-unicode

**Comment by blaxter** [comment]
tbh, It doesn't really help to have this pr open and the other as dependency of this one. The objective to open another pr with the unittest was to expedite the rtf fixes.

**Comment by javierag** [comment]
What way do you think will be better to speed up this?

**Comment by blaxter** [comment]
A pr without unittest, only the changes on RTFParser.* files (I can execute the unittest locally)

**Comment by javierag** [comment]
ok, so i  have to extract the unittest from the other Pr and put them there?.
I will need also to put a change in RFTHandler that is need for font table unittest (the method to mangle internal state). Is this ok?.

**Comment by blaxter** [comment]
ok for me

**Comment by javierag** [comment]
Ok, extraction done. You can review the new added tests.

File: OpenChange/RTFHandler.m [hunk 1]
```diff
@@ -293,6 +294,16 @@ - (void) dealloc
   [super dealloc];
 }
 
+- (NSString *) description
+{
+  NSString *description;
+  description = [NSString stringWithFormat:
+                          @"%u name=%@ family=%@family charset=%@ pitch=%u",
```
Comment (sixstone-qq) [inline OpenChange/RTFHandler.m]: This change conflicts with the #209 changeset. I thing the other one is the correct one, could you fix this svp?

File: OpenChange/RTFHandler.m [hunk 2]
```diff
@@ -332,6 +343,23 @@ - (RTFFontInfo *) fontInfoAtIndex: (unsigned int) theIndex
   return NSMapGet(fontInfos, key);
 }
 
+- (NSString *) description
+{
+  NSMutableString *description;
+  NSEnumerator *enumerator;
+  RTFFontInfo *fontInfo;
+
+  description = [NSMutableString stringWithFormat: @"Number of fonts:%u\n", [fontInfos count]];
```
Comment (sixstone-qq) [inline OpenChange/RTFHandler.m]: The same applies here...

Comment (javierag) [inline OpenChange/RTFHandler.m]: When one of this two pulls is merged, I will rebase the other and fix the conflicts.

**Comment by javierag** [comment]
I have rebased and fixed the conflicts with the description methods

**Comment by blaxter** [comment]
I'll review this (mostly my commits which where made without the intention of being merged to master) whenever I have some time. I'd like to finish first other things and I think this isn't a blocker so it won't be soon.

**Comment by sixstone-qq** [comment]
@blaxter I'll work on having these unit tests running in our CI system, so pass me over the issue once you finish.

**Comment by javierag** [comment]
@blaxter since the rtfhandler has been so deeply changed this commits are very useful as regression test so I wil let them.

**Comment by blaxter** [comment]
I've made some changes and I've `push --force` this branch: polishing the first commits, fixing `TestNGMailAddressParser` test (was failing because of bad expected strings) and removing `TestNGMimeAddressHeaderFieldGenerator` because values that were already q-encoded were being q-encoded again (bug?, feature?, no possible?). Without a real failure on this part (client showing bad headers and alike) I'd rather not spend more time on that.

**Comment by blaxter** [comment]
It would be easy to introduce a method that outputs junit xml output, if @sixstone-qq wants to do it, be my guest but this pr is ready to merge from my pov.

**Comment by javierag** [comment]
>>  because values that were already q-encoded were being q-encoded again (bug?, feature?, no possible?). 

If I remember correctly the NGMimeAddressHeaderFieldGenerator assummes that they are not encoded. But since is a free-text entry I would let things as they are; after all the suer can enter whatever wants.

File: Tests/Unit/GNUmakefile [hunk 3]
```diff
@@ -21,19 +21,23 @@ $(TEST_TOOL)_OBJC_FILES += \
...
 	TestNSString+Crypto.m \
 	TestNSString+URLEscaping.m \
 	TestNSString+Utilities.m \
-	TestNGMailAddressParser.m
+	TestNGMailAddressParser.m \
+	\
+	TestRTFHandler.m \
+	../../OpenChange/RTFHandler.m
+    # I don't know how to link against -l:SOGoBackend \
+	     undefined reference to `__objc_class_name_SOGoMailFolder'
```
Comment (sixstone-qq) [inline Tests/Unit/GNUmakefile]: The way to compile this is using Bundles... too much effort

**Comment by blaxter** [comment]
I've added test related with https://github.com/zentyal/sogo/pull/227

**Pull request merged**